### PR TITLE
layoutV3

### PR DESCRIPTION
**Pull request by gqio**

### CODE REVIEW

File: mdx-layout/src/index.tsx [hunk 1]
```diff
@@ -1,15 +1,31 @@
 import './markdown-github.scss';
 import { NavBar } from './NavBar';
 import React from 'react';
+import { FooterNav } from './FooterNav';
+import { DepGraph } from 'dependency-graph';
 
-export const Layout = ({ __config, children, ...props }) => {
-  const navbarProps = { ...__config, ...props };
+const buildGraph = (pages: any[]) => {
```
Comment (georges-gomes) [inline mdx-layout/src/index.tsx:7]: The typing of Pages here would be nice for typing but for education purpose.

**Review by georges-gomes — APPROVED** [review]

**Comment by georges-gomes** [comment]
The `Page` type would need to be review

**Comment by gqio** [comment]
> The `Page` type would need to be review

https://components.studio/edit/EsrEhWyXlBBcGqZJA47J/src/index.ts

**Comment by georges-gomes** [comment]
Page type:

Don't hesitate to comment the model for documentation.

Content and layout is not clear for me in the MDX context.

We should add the frontmatters (and all exports?) to it so we can have addition data flying into the navbar ?

**Comment by gqio** [comment]
> Page type:
> 
> Don't hesitate to comment the model for documentation.

Done, but I don't know why typings are not working properly 

> 
> Content and layout is not clear for me in the MDX context.

Content is the partial markup to render the mdx/mdjs/html (could be live, or static markup, as of today only live on mdj and mdjs)

Layout, is a shell and can be any template/document needed to wrap pages (like `App.html` in nuxt https://nuxtjs.org/docs/2.x/concepts/views#document-apphtml, `_document.js` in Next https://nextjs.org/docs/advanced-features/custom-document , `html.js` in gatsby https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/default-html.js)
Can also be used to template html themselves and or mdjs.

> 
> We should add the frontmatters (and all exports?) to it so we can have addition data flying into the navbar ?

All frontmatters are already reexported inside the mdx context to be able feed the layout (and the mdxcontent)

**Pull request merged**